examples/zero_inflated_poisson.py

```diff
@@ -0,0 +1,165 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Zero-Inflated Poisson regression model
+================================================
+
+We evaluate Zero-Inflated Poisson regression model with MAP, VI, and MCMC in terms of their predictive
+performances of the number of fish caught.
```

Comment:
Thanks, @wataruhashimoto52! Could you say a bit about the dataset here and define the problem here (e.g. Here we want to predict the number of fish caught given ... - I'm not sure)?

Looking like you have plots in your script, could you add one similar to [here](https://github.com/pyro-ppl/numpyro/blob/9719466a584af80683a7aa04401d4e065cc569ae/examples/funnel.py#L25) and [this folder](https://github.com/pyro-ppl/numpyro/tree/master/docs/source/_static/img/examples)? In addition, to have this example shown in the docs, you'll need to add an entry for it [in the applications section](https://github.com/pyro-ppl/numpyro/blob/master/docs/source/index.rst).